README.md

```diff
@@ -383,47 +386,76 @@ Which results in the following output:
 ### Working with Generic ASN.1 Structures
 
 The magic shown above is based on a from-scratch 100% KMP implementation of an ASN.1 encoder and parser.
-To parse any DER-encoded ASN.1 structure, call `Asn1Element.parse(derBytes)`, which will result in exactly a single
-`Asn1Element`.
-It can be re-encoded (and yes, it is a true re-encoding, since the original bytes are discarded after decoding) by
-accessing the lazily evaluated `.derEncoded` property.
+To parse any DER-encoded ASN.1 structure, call either:
+
+* `Asn1Element.parse(derBytes)`, which will consume all bytes and return the first parsed ASN.1 element.
+This method throws if more than a single toplevel ASN.1 Element it found or if any parsing errors occur.
+* `Asn1Element.parseFirst(byteIterator)`, which will try to parse a single toplevel ASN.1 element.
+Any remaining bytes can still be consumed from the iterator, as it will only be advanced to right after the frist parsed element.
+* `Asn1Element.parseAll(byteIterator)`, wich consumes all bytes, parses all toplevel ASN.1 elements, and returns them as list.
```

Comment:
this should have a `parse(derBytes)` overload

Reply:
done